### PR TITLE
Fix first 101 floating-promises

### PR DESCRIPTION
**Pull request by samholmes**
### CHANGELOG

 none

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204848238190307

### CODE REVIEW

File: src/actions/DeepLinkingActions.ts
Comment (swansontec) [inline src/actions/DeepLinkingActions.ts:134]: The `handleLink` function should simply `await` these lines so the error flows out.

Then, we should put catch statements on the places where `handleLink` is being called:

- `SwapDetailsTile` will probably `showError`
- `retryPendingDeepLink` should log the problem and leave the link un-handled
- `launchDeepLink` should probably become `async`, so it can `await` the `handleLink`. This kicks the can up to the `launchDeepLink` callers.

File: src/actions/DeepLinkingActions.ts
Comment (swansontec) [inline src/actions/DeepLinkingActions.ts:140]: `await` instead of `catch`

File: src/actions/DeepLinkingActions.ts
Comment (swansontec) [inline src/actions/DeepLinkingActions.ts:181]: `await` instead of `catch`

File: src/actions/DeepLinkingActions.ts
Comment (swansontec) [inline src/actions/DeepLinkingActions.ts:213]: `await` instead of `catch`

File: src/actions/DeepLinkingActions.ts
Comment (swansontec) [inline src/actions/DeepLinkingActions.ts:241]: `await` instead of `catch`

File: src/components/services/AirshipInstance.tsx
Comment (swansontec) [inline src/components/services/AirshipInstance.tsx:27]: Do we want the red box of doom in here?

File: src/actions/FioActions.tsx
Comment (swansontec) [inline src/actions/FioActions.tsx:67]: This should be turned into a `while` loop, using `await snooze(400)`, to delay the rest of the function until the FIO wallets appear.

File: src/actions/FioActions.tsx
Comment (swansontec) [inline src/actions/FioActions.tsx:78]: `await` instead of `catch`

File: src/actions/FioActions.tsx
Comment (swansontec) [inline src/actions/FioActions.tsx:92]: `await` instead of `catch`

File: src/actions/FioActions.tsx
Comment (swansontec) [inline src/actions/FioActions.tsx:147]: `await` instead of `catch`

File: src/actions/FioActions.tsx
Comment (swansontec) [inline src/actions/FioActions.tsx:175]: `await` instead of `catch`

File: src/actions/CreateWalletActions.tsx
Comment (swansontec) [inline src/actions/CreateWalletActions.tsx:233]: Should the `catch` go on `saveTxMetadata`, so we still navigate to the right place, even if we fail to save the tx details?

**Review by swansontec — CHANGES_REQUESTED** [review]

File: src/actions/AccountReferralActions.ts
Comment (samholmes) [reply src/actions/AccountReferralActions.ts:129]: I assumed these `saveAccountReferral` calls were concurrent with the redux action routines. If that's not the case, then these were bugs and I will add `await` instead. I was under the impression that these calls were more like the log calls; they shouldn't block. I will look more closely at each call to determine whether they should be concurrent like log or not concurrent (if they're bugs).

My main concern with a lot of "bug fixes" is that they may crop up with a lot of P0s (bugs that existed but never were apparent). That's a good thing, but could impact our sprint, time-wise.

File: src/util/logger.ts
Comment (samholmes) [reply src/util/logger.ts:161]: use `console.warn`

File: src/actions/LoginActions.tsx
Comment (samholmes) [reply src/actions/LoginActions.tsx:324]: I'll look into this one more. Question is whether it should be non-async and log it's own errors like `log`, or is it different because it's revenue related and we want to block on this async call.

File: src/actions/DeepLinkingActions.ts
Comment (samholmes) [reply src/actions/DeepLinkingActions.ts:134]: How is `handleLink` being called by `SwapDetailsTile`; I couldn't make that connection?

I implemented `retryPendingDeepLink` such that it catches internally; though correct me if I'm wrong, do we need a fail rate counter so we can avoid thrashing this retry routine?

File: src/components/services/AirshipInstance.tsx
Comment (samholmes) [reply src/components/services/AirshipInstance.tsx:27]: We really shouldn't be erroring here, so I'd argue this should be in devs faces if it were to ever fail.

File: src/actions/FioActions.tsx
Comment (samholmes) [reply src/actions/FioActions.tsx:67]: No while loop is needed since this function is recursive.

File: src/actions/TrackingActions.ts
Comment (swansontec) [inline src/actions/TrackingActions.ts:41]: Awesome. However, I wonder if this should have been in the previous commit.

File: src/components/scenes/SwapSettingsScene.tsx
Comment (swansontec) [inline src/components/scenes/SwapSettingsScene.tsx:313]: Awesome. It's always a good sign when we do `await dispatch`. It means we have correctly piped the promise up to the top of the call stack. 👍

File: src/actions/TrackingActions.ts
Comment (swansontec) [inline src/actions/TrackingActions.ts:17]: While we are in here, should this be `ThunkAction<void>`? Seems weird to return `unknown`.

File: src/components/services/AirshipInstance.tsx
Comment (swansontec) [reply src/components/services/AirshipInstance.tsx:27]: Fair enough.

File: src/actions/FioActions.tsx
Comment (swansontec) [inline src/actions/FioActions.tsx:66]: This will will be an infinite recursion. In the old version, they had a `return` statement to break the loop, but that's missing here.

Instead, try:

```ts
let account: EdgeAccount
let FioWallets: string[]

// Loop until the account and wallets appear (the app may not have loaded them yet):
let loopCount = 0
while (true) {
  account = state.core.account
  fioWallets = state.ui.wallets.fioWallets
  if (account != null && account.currencyConfig != null && fioWallets.length >0) break
  if (loopCount++ > MAX_OBT_DATA_CHECKS) return
  await snooze(400)
}
```

**Review by swansontec — CHANGES_REQUESTED** [review]
This looks much, much better! I am really happy with this approach.

Unfortunately, I did notice a missing return statement in the OBT check, so we need to get that fixed before merging.

**Review by swansontec — APPROVED** [review]

**Pull request merged**
